src/celeritas/geo/detail/BoundaryAction.cc

```diff
@@ -3,34 +3,43 @@
 // See the top-level COPYRIGHT file for details.
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 //---------------------------------------------------------------------------//
-//! \file celeritas/geo/generated/BoundaryAction.cc
-//! \note Auto-generated by gen-action.py: DO NOT MODIFY!
+//! \file celeritas/geo/detail/BoundaryAction.cc
 //---------------------------------------------------------------------------//
 #include "BoundaryAction.hh"
 
-#include <utility>
+#include <string>
 
 #include "corecel/Types.hh"
 #include "celeritas/global/ActionLauncher.hh"
+#include "celeritas/global/CoreParams.hh"
+#include "celeritas/global/CoreState.hh"
 #include "celeritas/global/TrackExecutor.hh"
-#include "../detail/BoundaryActionImpl.hh" // IWYU pragma: associated
+
+#include "BoundaryExecutor.hh"  // IWYU pragma: associated
 
 namespace celeritas
 {
-namespace generated
+namespace detail
+{
+
+BoundaryAction::BoundaryAction(ActionId aid,
+                               std::string label,
+                               std::string description)
+    : ConcreteAction(aid, label, description)
```

Comment:
Instead of passing in the label and description, can you move them here from CoreParams so that they're associated with this particular action?